src/systems/joint_trajectory_controller/JointTrajectoryController.cc

```diff
@@ -0,0 +1,792 @@
+      }
+    }
+
+    // Update the target for each joint that is defined in the trajectory, if needed
+    if (isTargetUpdateRequired && this->dataPtr->trajectory.status != Trajectory::Reached)
+    {
+      const auto targetPoint = this->dataPtr->trajectory.points[this->dataPtr->trajectory
+                                                                    .pointIndex];
+      for (auto jointIndex = 0u; jointIndex < this->dataPtr->trajectory.jointNames.size();
+           ++jointIndex)
+      {
+        const auto jointName = this->dataPtr->trajectory.jointNames[jointIndex];
+        if (this->dataPtr->actuatedJoints.count(jointName) == 0)
+        {
+          // Warning about unconfigured joint is already logged above
+          continue;
+        }
+        auto *joint = &this->dataPtr->actuatedJoints[jointName];
+        joint->SetTarget(targetPoint, jointIndex);
+
+        // Reset also the PID error of the affected joints
+        joint->ResetPIDs();
+      }
+
+      // If there are no more points after the current one, set the trajectory to Reached
+      if (this->dataPtr->trajectory.IsGoalReached())
+      {
+        this->dataPtr->trajectory.status = Trajectory::Reached;
+      }
+
+      // Publish current progress of the trajectory
+      ignition::msgs::Float progressMsg;
+      progressMsg.set_data(this->dataPtr->trajectory.ComputeProgress());
+      this->dataPtr->progressPub.Publish(progressMsg);
+    }
+  }
+
+  // Control loop
+  for (auto &actuatedJoint : this->dataPtr->actuatedJoints)
+  {
+    auto *joint = &actuatedJoint.second;
+    joint->Update(_ecm, _info.dt);
+  }
+}
+
+////////////////////////////////////////
+/// JointTrajectoryControllerPrivate ///
+////////////////////////////////////////
+
+void JointTrajectoryControllerPrivate::ConfigureJoint(
+    const Entity &_entity,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    const ignition::gazebo::EntityComponentManager &_ecm,
+    const std::vector<std::string> &_enabledJoints)
+{
+  const auto jointName = _ecm.Component<components::Name>(_entity)->Data();
+  auto jointIndex = this->actuatedJoints.size();
+
+  // Ignore duplicate joints
+  for (const auto &actuatedJoint : this->actuatedJoints)
+  {
+    if (actuatedJoint.second.entity == _entity)
+    {
+      ignwarn << "[JointTrajectoryController] Ignoring duplicate joint [" << jointName << "(Entity="
+              << _entity << ")]\".\n";
+      continue;
+    }
+  }
+
+  // Make sure the joint type is supported, i.e. it has a single actuated axis
+  const auto *jointType = _ecm.Component<components::JointType>(_entity);
+  switch (jointType->Data())
+  {
+  case sdf::JointType::PRISMATIC:
+  case sdf::JointType::REVOLUTE:
+  case sdf::JointType::CONTINUOUS:
+  case sdf::JointType::GEARBOX:
+  {
+    // Supported joint type
+    break;
+  }
+  case sdf::JointType::FIXED:
+  {
+    igndbg << "[JointTrajectoryController] Fixed joint [" << jointName << "(Entity=" << _entity
```

Comment:
I'm curious, why not make this a warning like the ones below?

Reply:
I just thought that it never makes sense to actuate a fixed joint, hence I selected debug instead of warning to reduce "disturbing" log clutter for models that are e.g. fixated to world.